internal/web_api.go

```diff
@@ -33,6 +34,7 @@ type api struct {
 	host                 string
 	customHeaders        http.Header
 	retryStrategy        RetryStrategyFunc
+	s                    *stats
```

Comment:
IMO this `stats` struct is not needed, doesn't provide any value. `sampleChan` could be member of `api` struct and `sample` it's function